flutter/shell/platform/tizen/tizen_autofill.cc

```diff
@@ -0,0 +1,188 @@
+// Copyright 2023 Samsung Electronics Co., Ltd. All rights reserved.
+// Use of this source code is governed by a BSD-style license that can be
+// found in the LICENSE file.
+
+#include "flutter/shell/platform/tizen/tizen_autofill.h"
+
+#include <app_common.h>
+#include <autofill_common.h>
+
+#include <functional>
+
+#include "flutter/shell/platform/tizen/logger.h"
+
+TizenAutofill::TizenAutofill() {
+  InitailizeAutofill();
+}
+
+TizenAutofill::~TizenAutofill() {
+  autofill_destroy(autofill_);
+}
+
+void TizenAutofill::InitailizeAutofill() {
+  autofill_create(&autofill_);
+
+  int ret = autofill_connect(
+      autofill_,
+      [](autofill_h autofill, autofill_connection_status_e status,
+         void* user_data) {},
```

Comment:
Is this deliberately no-op? No need to check the current connection status?

Reply:
`connected_` have been added to check connection status.

Reply:
`is_connected_` is usually better than just `connected_`.

https://dart.dev/guides/language/effective-dart/design#prefer-a-non-imperative-verb-phrase-for-a-boolean-property-or-variable

Reply:
You shouldn't need the extra `SetConnected` method. You can directly access private member fields via `user_data`.

Also you shouldn't assume the `status` is always `AUTOFILL_CONNECTION_STATUS_CONNECTED`.

Reply:
The `if (!connected_)` check will fail if it comes immediately after `Initailize()` and `autofill_connect` is not a synchronous call.

Related to the comment https://github.com/flutter-tizen/embedder/pull/25#discussion_r1105334997.

Reply:
> `is_connected_` is usually better than just `connected_`.

Modified to use `is_connected_`.
> You shouldn't need the extra `SetConnected` method. You can directly access private member fields via `user_data`.

Can I access the private region by casting `user_data` to `TizenAutofill`?
> Also you shouldn't assume the `status` is always `AUTOFILL_CONNECTION_STATUS_CONNECTED`.

Changed to call `SetConnected` according to `state`.
> The if (!connected_) check will fail if it comes immediately after Initailize() and autofill_connect is not a synchronous call.

`Autofill` will not work until connected. So we need to make sure it's connected.

Reply:
> Can I access the private region by casting `user_data` to `TizenAutofill`?

Yes, you can. Also please do not rename `user_data` to `data` unless it is appropriate. We often use `self` as the name of the casted pointer.

Reply:
Thank you for advice.
The `data` was changed to `user_data` again and use `self` as the name of the casted pointer.